persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/GenericAssayMapper.xml

```diff
@@ -9,6 +9,28 @@
                 <result property="value" column="VALUE"/> 
     </resultMap>
 
+    <select id="getGenericAssayMetaByGenericAssayType" resultType="org.cbioportal.model.meta.GenericAssayMeta">	
+        SELECT DISTINCT genetic_entity.STABLE_ID as stableId,
+        genetic_entity.ENTITY_TYPE as entityType
+        FROM genetic_entity
+        <if test="genericAssayType != null">	
+                LEFT JOIN	
+            genetic_alteration AS ga ON ga.GENETIC_ENTITY_ID = genetic_entity.ID	
```

Comment:
I'm not following this - why join genetic_alteration (has data) or genetic_profile below.  Don't you need to join generic_entity_properties table instead?